File: Java/IntersectionOfArray.java
import java.util.HashSet;
import java.util.Arrays;

public class IntersectionOfArray {

    public static int[] solution(int[] nums1, int[] nums2) {
        HashSet<Integer> set1 = new HashSet<>();
        HashSet<Integer> resultSet = new HashSet<>();

        // Add elements of nums1 to set1
        for (int num : nums1) {
            set1.add(num);
        }

        // Check for intersection with nums2
        for (int num : nums2) {
            if (set1.contains(num)) {
                resultSet.add(num);
            }
        }

        // Convert resultSet to an array
        int[] result = new int[resultSet.size()];
        int i = 0;
        for (int num : resultSet) {
            result[i++] = num;
        }
        return result;
    }

    public static void main(String[] args) {
        int[] nums1 = {1, 2, 2, 1};
        int[] nums2 = {2, 2};

        // Corrected method call and added Arrays.toString() for proper printing
        System.out.println(Arrays.toString(solution(nums1, nums2))); 
    }
}

File: Python/IntersectionOfArray.py
class Solution(object):
    def intersection(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        return list(set(nums1) & set(nums2))  # Using set intersection

# Example usage:
sol = Solution()
print(sol.intersection([1,2,2,1], [2,2]))  # Output: [2]
print(sol.intersection([4,9,5], [9,4,9,8,4]))  # Output: [9,4] (order may vary)
